.github/workflows/test.yml

```
@@ -21,17 +21,20 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Before install
-      run: |
-        python3 -m venv ~/venv
-        source ~/venv/bin/activate
-
     - name: Install packages
+      shell: bash
       run: |
-        pip install --upgrade pip wheel setuptools
-        pip install -r requirements.txt
-        pip install .
-        pip list
+        if [ "$RUNNER_OS" == "Windows" ]; then 
+          python -m pip install --upgrade pip wheel setuptools
+          python -m pip install -r requirements.txt
+          python -m pip install .
+          python -m pip list
+        else
+          pip install --upgrade pip wheel setuptools
+          pip install -r requirements.txt
+          pip install .
+          pip list
+        fi
```

```suggestion
        python -m pip install --upgrade pip wheel setuptools
        python -m pip install -r requirements.txt
        python -m pip install .
        python -m pip list
```
I think you can get away with the pip module invocation on all platforms which would get rid of the conditional (unless you tried this on mac/linux already and something went wrong). Not critical though.

---

Windows won't let you update pip when it is running.

---

Never mind, I misread what you wrote.